Gimbalize/src/timeSync.h

```diff
@@ -0,0 +1,29 @@
+#ifndef TIMESYNC_H
+#define TIMESYNC_H
+
+// Raghav: I do not have the dependencies for this right now but it should work
+#include <vector>
+#include <array>
+#include <utility>
+#include <algorithm>
+
+class TimeSync {
+public:
+    // Constructor: max_history_size controls how many readings are stored
+    TimeSync(size_t max_history_size = 100);
+
+    // Add a new sensor reading (timestamp in ms and values as array)
+    void addReading(double timestamp, const std::array<double, 3>& values);
+
+    // Get interpolated values at the desired time
+    std::array<double, 3> getValuesAt(double desired_time) const;
```

Comment:
No edge-case handling for when time is 0 for instance (will return null), cover all edge cases